org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/XtextFormatterTest.java

```diff
@@ -0,0 +1,190 @@
+		String model =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	  '->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=>('>' '>') | '>')\n" +
+				"	| '<' (=>('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';\n" +
+				"		";
+		String expected =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	'->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=> ('>' '>') | '>')\n" +
+				"	| '<' (=> ('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';";
+		assertFormattedAs(Strings.toPlatformLineSeparator(model), expected);
+	}
+
+	@Test
+	public void formatting_with_unix_line_endings01() throws Exception {
+		String model =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	  '->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=>('>' '>') | '>')\n" +
+				"	| '<' (=>('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';";
+		String expected =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	'->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=> ('>' '>') | '>')\n" +
+				"	| '<' (=> ('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';";
+		assertFormattedAs(Strings.toUnixLineSeparator(model), expected);
+	}
+
+	@Test
+	public void formatting_with_unix_line_endings02() throws Exception {
+		String model =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	  '->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=>('>' '>') | '>')\n" +
+				"	| '<' (=>('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';\n";
+		String expected =
+				"grammar org.xtext.example.mydsl.MyDsl\n" +
+				"\n" +
+				"generate myDsl 'http://www.xtext.org/example/mydsl/MyDsl'\n" +
+				"\n" +
+				"OpOther:\n" +
+				"	'->'\n" +
+				"	| '..<'\n" +
+				"	| '>' '..'\n" +
+				"	| '..'\n" +
+				"	| '=>'\n" +
+				"	| '>' (=> ('>' '>') | '>')\n" +
+				"	| '<' (=> ('<' '<') | '<' | '=>')\n" +
+				"	| '<>'\n" +
+				"	| '?:';";
+		assertFormattedAs(Strings.toUnixLineSeparator(model), expected);
+	}
+
+	private void assertFormattedAs(CharSequence input, CharSequence expected) throws Exception {
+		Assert.assertEquals(Strings.toPlatformLineSeparator(expected).toString(), formattedText(input));
```

Comment:
It's not clear to me why we use `toUnixLineSeparator` for input and  `toPlatformLineSeparator` for expected.

Reply:
this is at it was in the original :(